cpp/src/arrow/compute/row/grouper.h

```diff
@@ -30,8 +30,15 @@
 namespace arrow {
 namespace compute {
 
-/// \brief A segment of contiguous rows for grouping
-struct ARROW_EXPORT GroupingSegment {
+/// \brief A segment.
+/// A segment group is a chunk of continous rows that has the same segment key. (For
+/// example, in ordered time series processing, segment key can be "date", and a segment
+/// group can be all the rows that belong to the same date.) A segment group can span
+/// across multiple exec batches. A segment is a chunk of continous rows that has the same
+/// segment key within a given batch. When a ? span cross batches, it will have multiple
```

Comment:
Looks like the question mark here is a typo?

Reply:
Fixed.